tests/cpp/dataLevel0BlocksMemory_test.cpp

```diff
@@ -0,0 +1,175 @@
+    size_t batch_num = (total_items / (num_threads <= 1 ? 1 : (num_threads - 1))) + 1;
+    auto start_time = std::chrono::system_clock::now();
+    for (size_t idx = 0; idx < total_items; idx += batch_num) {
+        size_t start = idx;
+        size_t end = std::min(idx + batch_num, total_items);
+        threads.push_back(
+            std::thread(
+                [alg_hnsw, start, end] {
+                    for (size_t i = start; i < end; i++) {
+                       alg_hnsw->addPoint(data.data() + i * dimension, i);
+                    }
+                }
+            )
+        );
+    }
+    for (auto &thread : threads) {
+       thread.join();
+    }
+    threads.clear();
+    auto end_time = std::chrono::system_clock::now();
+    double duration_in_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time).count();
+    double duration_in_seconds = static_cast<double>((std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time)).count()) / 1000.0;
+    size_t qps = (duration_in_seconds == 0 ? total_items : total_items / duration_in_seconds);
+    double latency = (total_items == 0 ? 0 : duration_in_ms / total_items);
+    std::cout << "Start " << num_threads << " thread to add " << total_items << " items to hnsw index, cost "
+              << duration_in_seconds << " seconds, qps: " << qps  << ", latency: " << latency << "ms" << std::endl;
+
+
+    num_threads = (num_query >= max_thread_num ? max_thread_num : num_query);
+    batch_num = (num_query / (num_threads <= 1 ? 1 : (num_threads - 1))) + 1;
+    start_time = std::chrono::system_clock::now();
+    for (size_t idx = 0; idx < num_query; idx += batch_num) {
+        size_t start = idx;
+        size_t end = std::min(idx + batch_num, num_query);
+        threads.push_back(
+            std::thread(
+                [alg_hnsw, start, end] {
+                    for (size_t i = start; i < end; i++) {
+                        const void* p = query.data() + i * dimension;
+                        auto gd = alg_hnsw->searchKnn(p, topk);
+                    }
+                }
+            )
+        );
+    }
+    for (auto &thread : threads) {
+       thread.join();
+    }
+    threads.clear();
+    end_time = std::chrono::system_clock::now();
+    duration_in_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time).count();
+    duration_in_seconds = static_cast<double>((std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time)).count()) / 1000.0;
+    qps = (duration_in_seconds == 0 ? num_query : num_query / duration_in_seconds);
+    latency = (num_query == 0 ? 0 : duration_in_ms / num_query);
+    std::cout << "Start " << num_threads << " thread to exec " << num_query << " searchKnn, cost "
+              << duration_in_seconds << " seconds, qps: " << qps  << ", latency: " << latency << "ms" << std::endl;
+
+    delete alg_hnsw;
+}
+
+}  // namespace
+
+int main() {
```

Comment:
Could you please add a test to check the `resizeIndex` function as well?
For example, we can call `resizeIndex` multiple times and then verify that the recall is high and all elements' data is valid in the index.